src/Execution/Executor.php

```diff
@@ -379,7 +385,7 @@ protected function executeFields(
                 continue;
             }
 
-            $doesContainPromise = $doesContainPromise || $this->isPromise($result);
+            $doesContainPromise = $doesContainPromise || $result instanceof ExtendedPromiseInterface;
```

Comment:
@crisu83 Not sure why you don't like the `isPromise` method, but we use it in ~3 places at least.

Reply:
Because it’s better to use `instanceof` instead of the `@var` comments.

Reply:
@crisu83 If you insist, feel free to change, but we actually don't really use it in the code.